routes/account.js

```diff
@@ -80,6 +83,86 @@ router.post('/logout', function(req, res) {
     res.sendStatus(200);
 });
 
+router.get('/updateProfile', verifyJWTCookie, function(req, res) {
+    UserModel.findOne({id: req.user.id}, function(err, user) {
+        if (err) {
+            return res.status(422).json( {success: false});
+        }
+        if (user) {
+            res.render('updateProfile', {username: user.username, 
+                                           firstname: user.firstname, 
+                                           lastname: user.lastname,
+                                           email: user.email, 
+                                           password: user.password})
+            
+        } else {
+            res.status(401).json( {success: false, invalidCredentials: true});
+        }
+    });
+});
+
+router.post('/updateProfile', verifyJWTCookie, function(req, res) {
+    var updateUser = new UserModel();
```

Comment:
It isn't necessary to make a new model object here. You can just use the `user` variable in the callback to `findOne` and modify it directly. Then when you are done modifying it, call `save` on it.

Reply:
Removed line 105